src/test/java/io/vertx/core/DeploymentTest.java

```diff
@@ -1234,6 +1234,23 @@ public void testUndeployAll() throws Exception {
     vertx = null;
   }
 
+  @Test
+  public void testDeployChildOnParentUndeploy() throws Exception {
+    class ParentVerticle extends AbstractVerticle {
+      @Override
+      public void stop(Promise<Void> stopPromise) {
+        vertx.deployVerticle(ChildVerticle.class.getName())
+          .<Void>mapEmpty()
+          .setHandler(stopPromise);
+      }
+    }
+
+    CountDownLatch latch = new CountDownLatch(1);
+    vertx.deployVerticle(new ParentVerticle(), onSuccess(id ->
+      vertx.undeploy(id, onFailure(u -> latch.countDown()))));
+    awaitLatch(latch);
```

Comment:
instead use `await()` and `testComplete()` to signal test completion

Reply:
Done.Do you want me to squash to single commit?

Reply:
thanks, no I'll do it on merge